sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -159,11 +159,19 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       plan: LogicalPlan,
       cascade: Boolean,
       blocking: Boolean = false): Unit = {
+    val analyzedPlan = if (plan.resolved) {
+      plan
+    } else {
+      val qe = spark.sessionState.executePlan(plan)
+      qe.assertAnalyzed()
```

Comment:
where do we call `uncacheQuery` with an unresolved plan?

Reply:
```
sql("CACHE TABLE t1 AS SELECT * FROM testData")
sql("UNCACHE TABLE t1")
```
`t1` passed to `uncacheQuery` is:
```
'SubqueryAlias t1
+- View (`t1`, ['key,'value])
   +- 'Project [upcast('key, IntegerType) AS key#293, upcast('value, StringType) AS value#294]
      +- 'Project [*]
         +- 'UnresolvedRelation [testData], [], false
```
Now that temporary views are always resolved before they are created, we need to resolve `t1` in this case.

Note that we couldn't make the table of `CacheTable`/`UncacheTable` as a child due to https://github.com/apache/spark/pull/30598#discussion_r536384428.

Reply:
Thanks for the reminds. We can't make the table the child of `CacheTable/UncacheTable`, but we can resolve the table in the analyzer like what we did for INSERT?

Reply:
`CacheTable` is tricky as the query can be complex and needs to go through the entire analyzer, `UncacheTable` should be fine as we can just update a few rules to handle it, which is similar to INSERT.

Reply:
Actually, it was subtle why `UnresolvedRelation [testData]` inside temp view `t1` was not resolved.
When the temp view `t1` is looked up, its text is parsed and produces:
```
'SubqueryAlias t1
+- View (`t1`, ['key,'value])
   +- 'Project [upcast('key, IntegerType) AS key#293, upcast('value, StringType) AS value#294]
      +- 'Project [*]
         +- 'UnresolvedRelation [testData], [], false
```
Now this logical plan goes thru the following to resolve this view further: https://github.com/apache/spark/blob/e0053853c90d39ef6de9d59fb933525e20bae1fa/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L1100

However, we have the following check that doesn't allow resolving further:
https://github.com/apache/spark/blob/e0053853c90d39ef6de9d59fb933525e20bae1fa/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L924-L925

But, we are resolving a temp view in this context, so I updated the check to allow relations in temp views are resolved correctly.